Games/Conway/Game.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "Game.h"
+#include <LibCore/ConfigFile.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/Bitmap.h>
+#include <stdlib.h>
+#include <time.h>
+
+Game::Game()
+{
+    srand(time(nullptr));
+    reset();
+}
+
+Game::~Game()
+{
+}
+
+void Game::reset()
+{
+    stop_timer();
+    seed_universe();
+    start_timer(m_sleep);
+    update();
+}
+
+void Game::seed_universe()
+{
+    for (int y = 0; y < m_rows; y++)
+        for (int x = 0; x < m_columns; x++)
+            m_universe[y][x] = (arc4random() % 2) ? 1 : 0;
+}
+
+void Game::update_universe()
+{
+    bool new_universe[m_rows][m_columns];
+
+    int n;
+    for (int y = 0; y < m_rows; y++) {
+        for (int x = 0; x < m_columns; x++) {
+            n = 0;
+            int cell = m_universe[y][x];
+
+            for (int y1 = y - 1; y1 <= y + 1; y1++)
+                for (int x1 = x - 1; x1 <= x + 1; x1++)
+                    if (m_universe[(y1 + m_rows) % m_rows][(x1 + m_columns) % m_columns])
+                        n++;
+
+            if (cell)
+                n--;
+
+            if (n == 3 || (n == 2 && cell))
+                new_universe[y][x] = true;
+            else
+                new_universe[y][x] = false;
```

Comment:
`cell` should be `cell != 0` to more explicit.
 
Also, can't this just be:
```suggestion
            new_universe[y][x] = n == 3 || (n == 2 && cell != 0);
```

Reply:
> 
> 
> I think you should use `cell != 0` instead of `cell` here to be more explicit. :)

why

> Also, can't this just be: [...]

It could, if you like mashing together declarations and conditionals into a single line with 6 equals signs and multiple parenthesis. Seems like a terrible decision for no benefit.

Reply:
I mean, it's just one one group of parenthesis.
```cpp
new_universe[y][x] = n == 3 || (cell && n == 2)
```
I just don't like using this personally:
```cpp
bool var;
if (EXPRESSION)
    var = true;
else
    var = false;
```
since `var` it is an indirect way of writing:
```cpp
bool var = EXPRESSION;
```
But no big deal if you prefer writing it that way. This was just a comment.